### ISSUE TITLE
Stacktrace on master

### ISSUE THREAD

**Issue by eraserhd** [weavejester/lein-ring#149]
@MichaelBlume Here is the stack trace.

I can't post the project I have, but if this doesn't help, I can whittle down the project to the minimum thing which fails.  Let me know.

```
java.lang.UnsupportedOperationException: nth not supported on this type: PersistentArrayMap
 at clojure.lang.RT.nthFrom (RT.java:857)
    clojure.lang.RT.nth (RT.java:807)
    leiningen.jar$manifest_entry.invoke (jar.clj:26)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.AFn.applyTo (AFn.java:144)
    clojure.core$apply.invoke (core.clj:626)
    clojure.core$partial$fn__4228.doInvoke (core.clj:2468)
    clojure.lang.RestFn.invoke (RestFn.java:408)
    clojure.core$map$fn__4245.invoke (core.clj:2559)
    clojure.lang.LazySeq.sval (LazySeq.java:40)
    clojure.lang.LazySeq.seq (LazySeq.java:49)
    clojure.lang.Cons.next (Cons.java:39)
    clojure.lang.RT.next (RT.java:598)
    clojure.core$next.invoke (core.clj:64)
    clojure.string$join.invoke (string.clj:186)
    leiningen.jar$make_manifest.invoke (jar.clj:38)
    leiningen.ring.war$make_ring_manifest$fn__1371.invoke (war.clj:42)
    clojure.core$with_redefs_fn.invoke (core.clj:6861)
    leiningen.ring.war$make_ring_manifest.invoke (war.clj:41)
    leiningen.ring.war$create_war.invoke (war.clj:179)
    skeletor.plugin$create_war_hook.invoke (plugin.clj:17)
    clojure.lang.Var.invoke (Var.java:388)
    clojure.lang.AFn.applyToHelper (AFn.java:160)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.core$apply.invoke (core.clj:626)
    robert.hooke$compose_hooks$fn__11692.doInvoke (hooke.clj:40)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invoke (core.clj:624)
    robert.hooke$run_hooks.invoke (hooke.clj:46)
    robert.hooke$prepare_for_hooks$fn__11697$fn__11698.doInvoke (hooke.clj:54)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.lang.AFunction$1.doInvoke (AFunction.java:29)
    clojure.lang.RestFn.invoke (RestFn.java:421)
    leiningen.ring.war$write_war.invoke (war.clj:211)
    leiningen.ring.war$war.invoke (war.clj:241)
    leiningen.ring.war$war.invoke (war.clj:230)
    clojure.lang.AFn.applyToHelper (AFn.java:154)
    clojure.lang.AFn.applyTo (AFn.java:144)
    clojure.core$apply.invoke (core.clj:626)
    leiningen.ring$ring.doInvoke (ring.clj:27)
    clojure.lang.RestFn.invoke (RestFn.java:425)
    clojure.lang.Var.invoke (Var.java:383)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.core$apply.invoke (core.clj:626)
    leiningen.core.main$partial_task$fn__6071.doInvoke (main.clj:253)
    clojure.lang.RestFn.applyTo (RestFn.java:139)
    clojure.lang.AFunction$1.doInvoke (AFunction.java:29)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invoke (core.clj:626)
    leiningen.core.main$apply_task.invoke (main.clj:303)
    leiningen.core.main$resolve_and_apply.invoke (main.clj:309)
    leiningen.core.main$_main$fn__6136.invoke (main.clj:377)
    leiningen.core.main$_main.doInvoke (main.clj:366)
    clojure.lang.RestFn.invoke (RestFn.java:421)
    clojure.lang.Var.invoke (Var.java:383)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.core$apply.invoke (core.clj:624)
    clojure.main$main_opt.invoke (main.clj:315)
    clojure.main$main.doInvoke (main.clj:420)
    clojure.lang.RestFn.invoke (RestFn.java:457)
    clojure.lang.Var.invoke (Var.java:394)
    clojure.lang.AFn.applyToHelper (AFn.java:165)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.main.main (main.java:37)
```

**Comment by eraserhd** [comment]
Output of `lein -v`:

```
Leiningen 2.5.0 on Java 1.7.0_51 Java HotSpot(TM) 64-Bit Server V
```

**Comment by MichaelBlume** [comment]
Thanks, I was just about to ask for the version =)

**Comment by MichaelBlume** [comment]
Can you show me your :manifest entry in your project.clj? Like I can reproduce if your :manifest is a list of maps, but I don't think that's legal?

**Comment by eraserhd** [comment]
No `:manifest` keyword anywhere in the `project.clj`.

Plugins (in addition to lein-ring): lein-midje 0.3.0.  In `~/.lein/profiles.clj`: cider/cider-nrepl 0.7.0, lein-ancient 0.5.4, net.eraserhead.clojure/lein-release 1.0.6, and our internal plugin.  I get the same error if I comment out our plugin.

Use of internal plugin repositories.

**Comment by eraserhd** [comment]
Perhaps relevant - the configuration of the `:ring` key:
```
:ring {:handler points-api.core/app}
```

**Comment by MichaelBlume** [comment]
Thank you, that last one sorted it =)

Fix will be in soon, thanks again for the report.

**Comment by MichaelBlume** [comment]
ok, so the problem is here https://github.com/technomancy/leiningen/commit/bd6d25b32351 -- I developed #146 against leiningen 2.5.1 not realizing that they'd recently changed how they handled their default properties.  Also clojure.core/merge acts really weird if you call it with not-maps.

So what I'm *tempted* to do is just have the code look at the actual default values and do something like

```(into (empty lein-defaults) ring-defaults)``` 

and get a map if they're a map and a vector if they're a vector. 

But, well, *ultimately* this regression happened because I tried to interoperate with parts of leiningen marked private/internal. That is not a good strategy for compatibility with lots of leiningen versions, either past or future. So what I'm going to do instead is just copy/past the relevant functions into lein-ring and use our copies.
